test/mongodb.test.js

```diff
@@ -515,14 +515,30 @@ describe('mongodb connector', function() {
         posts.should.have.lengthOf(1);
         post = posts[0];
         post.should.have.property('title', 'a');
-        post.should.have.property('content', undefined);
+        should.not.exist(post.content);
         should.not.exist(post._id);
 
         done();
       });
     });
   });
 
+  it('all return should honor filter.fields with `_id` selected', function(done) {
+    var post = new PostWithObjectId({ title: 'a', content: 'AAA' });
+    post.save(function(err, post) {
+      PostWithObjectId.all({ fields: ['_id', 'content'], where: { title: 'a' }}, function(err, posts) {
+        should.not.exist(err);
```

Comment:
please return error here to print a more readable error message when test fails:
```js
if(err) return done(err);
```